2004-01-12  David Mosberger  <davidm@hpl.hp.com>
	    James E Wilson  <wilson@specifixinc.com>

	PR target/18987
	* config/ia64/ia64.c (process_set): For alloc insn, only call
	process_epilogue is !frame_pointer_needed.

http://gcc.gnu.org/bugzilla/show_bug.cgi?id=18987

--- gcc/gcc/config/ia64/ia64.c
+++ gcc/gcc/config/ia64/ia64.c
@@ -7840,7 +7840,7 @@
 	     followed by a new prologue.  If the procedure doesn't
 	     have a memory-stack frame, we'll issue a dummy ".restore
 	     sp" now.  */
-	  if (current_frame_info.total_size == 0)
+	  if (current_frame_info.total_size == 0 && !frame_pointer_needed)
 	    /* if haven't done process_epilogue() yet, do it now */
 	    process_epilogue ();
 	  fprintf (asm_out_file, "\t.prologue\n");
